lib/db_connection.ex

```diff
@@ -1085,6 +1087,32 @@ defmodule DBConnection do
     enum.(acc, fun)
   end
 
+  @doc false
+  def register_as_pool(conn_module) do
+    Process.put(@connection_module_key, conn_module)
+  end
+
+  @doc """
+  Returns connection module used by the given connection pool process.
+
+  If the given process is not a connection pool, `:error` is returned.
+  """
+  @spec connection_module(pid() | atom()) :: {:ok, module()} | :error
```

Comment:
This should probably be `conn()`, but I'm not sure if we can handle `%DBConnection{}` reasonably.